public_dropin_environments/python3_sklearn_fedramp_compliant/Dockerfile

```diff
@@ -0,0 +1,73 @@
+# This is a private chain-guard development image that is stored in DataRobot's private registry.
+# Replace it with your own development chain-gaurd image if you build your own.
+ARG BASE_ROOT_IMAGE=datarobotdev/mirror_chainguard_datarobot.com_python-fips:3.11-dev
+FROM ${BASE_ROOT_IMAGE} AS build
+
+USER root
+
+# Fix these CVEs: CVE-2025-0395, CVE-2024-13176, CVE-2025-0395
+RUN apk update && apk upgrade --available && \
+    apk add --no-cache glibc=2.40-r6 glibc-dev=2.40-r6 glibc-locale-posix=2.40-r6 \
+    ld-linux=2.40-r6 libcrypt1=2.40-r6 nss-db=2.40-r6 nss-hesiod=2.40-r6 \
+    openssl=3.4.0-r6 libcrypto3=3.4.0-r6 libssl3=3.4.0-r6 && \
+    rm -rf /var/cache/apk/* && \
+    mkdir -p /opt/code  # Required for custom-models
+
+USER nonroot
+
+ENV HOME=/home/nonroot
+WORKDIR ${HOME}
+
+COPY requirements.txt requirements.txt
+RUN pip install -U pip && \
```

Comment:
Why aren't you using a venv?

Reply:
Is it anyway a temporary stage, where stuff are being installed by the `non-root` user. 
The products are installed under the `non-root` user home directory and later being copied to the production image.
Do you think we should do it in `venv`?

Reply:
Ah, I _think_ I understand.  I think that `pip` automatically detects when it doesn't have permission to install packages into the system location, then it falls back to installing into users's homedir. You could make that a little more obvious here by just explicitly setting the `--user` flag to `pip install --user`. Then I'm fine with not using a venv.